plugins/outputs/nats/nats_test.go

```diff
@@ -1,46 +1,260 @@
 package nats
 
 import (
+	"context"
+	_ "embed"
 	"fmt"
+	"path/filepath"
+	"reflect"
 	"testing"
 
+	"github.com/docker/go-connections/nat"
+	"github.com/nats-io/nats.go/jetstream"
 	"github.com/stretchr/testify/require"
 	"github.com/testcontainers/testcontainers-go/wait"
 
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
 	"github.com/influxdata/telegraf/plugins/serializers/influx"
 	"github.com/influxdata/telegraf/testutil"
 )
 
-func TestConnectAndWriteIntegration(t *testing.T) {
+func TestConnectAndWriteNATSIntegration(t *testing.T) {
 	if testing.Short() {
 		t.Skip("Skipping integration test in short mode")
 	}
+	natsServicePort := "4222"
```

Comment:
I don't think it is a good idea to hard-code ports in tests, as they might fail in case the port is already in use when running the test.

Reply:
I think this was already present in the code previously. 
https://github.com/influxdata/telegraf/pull/14236/files#diff-112c292ad4b90deafc40e933ee4ba133549a94ccf40e6d43bcbaf2fc17614318L19

But I agree, this needs to change.

Reply:
Edit:
It turns out this port in inside the container and not for the host. The host port is assigned by the library after the mapping is done and we use that to communicate.